lxd/device/disk.go

```diff
@@ -461,24 +461,29 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
 				return fmt.Errorf("Failed to get storage pool %q: %w", d.config["pool"], err)
 			}
 
-			// Custom volume validation.
+			// Non-root volume validation.
 			if !instancetype.IsRootDiskDevice(d.config) {
+				volumeType, dbVolumeType, _, volumeName, err := storagePools.DiskVolumeSourceParse(d.config["source"])
+				if err != nil {
+					return err
+				}
+
 				// Derive the effective storage project name from the instance config's project.
-				storageProjectName, err = project.StorageVolumeProject(d.state.DB.Cluster, instConf.Project().Name, cluster.StoragePoolVolumeTypeCustom)
+				storageProjectName, err = project.StorageVolumeProject(d.state.DB.Cluster, instConf.Project().Name, dbVolumeType)
 				if err != nil {
 					return err
 				}
 
 				// GetStoragePoolVolume returns a volume with an empty Location field for remote drivers.
 				err = d.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-					dbCustomVolume, err = tx.GetStoragePoolVolume(ctx, d.pool.ID(), storageProjectName, cluster.StoragePoolVolumeTypeCustom, d.config["source"], true)
+					dbCustomVolume, err = tx.GetStoragePoolVolume(ctx, d.pool.ID(), storageProjectName, dbVolumeType, volumeName, true)
 					return err
 				})
 				if err != nil {
-					return fmt.Errorf("Failed loading custom volume: %w", err)
+					return fmt.Errorf("Failed loading \"%s/%s\" from project %q: %w", volumeType, volumeName, storageProjectName, err)
```

Comment:
Please can you use backticks for fmt.Errorf and then you can use double quotes unescaped for readability.